developer/src/common/web/utils/src/xml-utils.ts

```diff
@@ -0,0 +1,122 @@
+import { xml2js } from "./index.js";
+
+export class KeymanXMLOptions {
+    type: 'keyboard3'         // LDML <keyboard3>
+        | 'keyboard3-test'    // LDML <keyboardTest3>
+        | 'kps'   // <Package>
+        | 'kvks'            // <visualkeyboard>
+        | 'kpj' //    // <KeymanDeveloperProject>
+        ;
+}
+
+/** wrapper for XML parsing support */
+export class KeymanXMLReader {
+    public constructor(public options: KeymanXMLOptions) {
+    }
+
+    public parse(data: string): any {
+        const parser = this.parser();
+        let a: any;
+        parser.parseString(data, (e: unknown, r: unknown) => { if (e) throw e; a = r; });
+        return a;
+    }
+
+    public parser() {
+        const { type } = this.options;
+        switch (type) {
+            case 'keyboard3':
```

Comment:
This would be more elegant with a declarative approach, with const options object indexed on type, than as a honking big switch statement (and @markcsinclair will tell you off if you use `switch`). A better type in `PARSER_OPTIONS: any` may be available?

```
const PARSER_OPTIONS: {[index: KeymanXMLType]: any} = {
  'keyboard3': {
    explicitArray: false,
    mergeAttrs: true,
    includeWhiteChars: false,
    emptyTag: {} as any,
  },
  'keyboard3-test': {
    preserveChildrenOrder: true, // needed for test data
    explicitChildren: true, // needed for test data
  },
  ...
};

if(!PARSER_OPTIONS[type]) {
  throw...
}
return new xml2js.Parser(PARSER_OPTIONS[type]);
```